### PR TITLE
[KED-2169] Remove get_project_context()

### PR DESCRIPTION
**Pull request by lorenabalan**
## Motivation and Context
Why was this PR created?
`get_project_context()` has been there for backwards compatibility with Kedro 0.14.x Since we no longer support old versions <0.16, and the function will be deprecated, we can remove this.
Also updated some module paths.

❗ This change is compatible with both 0.16.x and 0.17.x.

## How has this been tested?
Installed kedro from `develop` branch.

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Assigned myself to the PR
- [ ] Added tests to cover my changes
- [ ] Added new entries to the `RELEASE.md` file

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: kedro_docker/plugin.py [hunk 1]
```diff
@@ -146,10 +149,18 @@ def docker_group():
...
+    project_path = Path.cwd()
     template_path = Path(__file__).parent / "template"
-    verbose = get_project_context("verbose")
+
+    if KEDRO_VERSION.match(">=0.17.0"):
+        verbose = KedroCliError.VERBOSE_ERROR  # pylint: disable=no-member
+    else:
+        from kedro.framework.cli.cli import (  # pylint: disable=import-outside-toplevel
+            _VERBOSE,
+        )
```
Comment (andrii-ivaniuk) [inline kedro_docker/plugin.py:160]: `KedroCliError.VERBOSE_ERROR` equal to `False` whereas `...cli._VERBOSE` equal to `True`. 

I believe `verbose` should be the same for both versions.

**Review by andrii-ivaniuk — APPROVED** [review]
LGTM

File: kedro_docker/plugin.py (hunk 1 above)
Comment (lorenabalan) [reply kedro_docker/plugin.py:160]: Mm that's a breaking change between 0.16 and 0.17, I'm afraid there's no way getting around that. This code matches what `get_project_context()` would return for `verbose` key.

Comment (DmitriiDeriabinQB) [reply kedro_docker/plugin.py:160]: Maybe `from kedro.framework.cli.cli import _VERBOSE as verbose`?

**Review by DmitriiDeriabinQB — APPROVED** [review]
LGTM! 👏 
Could you please also update the release notes to mention the removal of `get_project_context` usage due to its deprecation?

**Pull request merged**